### ISSUE TITLE
a command line option to register godot as a desktop application

### ISSUE THREAD

**Issue by O-XiXu** [godotengine/godot-proposals#2941]
### Describe the project you are working on

No project, except installing godot from my linux repository, i installed it from official web site
so it didn't get register

### Describe the problem or limitation you are having in your project

for registering the app to my desktop (to show app icon in applications menu) i had to do it manually

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

with adding a new option in command line for registering the app it take less time to setup godot

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

tor browser has something like this, --register-app and --unregister-app
this option add a .dekstop file in .local/share/applications (just for user)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

More options More free time to have More fun

### Is there a reason why this should be core and not an add-on in the asset library?

this can't be a add-on but user can register it manually, bad option for beginners,
and something else probably you need to automatically add a icon to a local directory and
remove it  with --unregister-app option

**Comment by Calinou** [comment]
This is not something Godot should be doing. It should be done using the distribution's package manager or an application distribution system such as [Flatpak](https://flatpak.org/).

For the record, it's already possible to package Godot projects as Flatpaks. Here's an example with [Pixelorama](https://flathub.org/apps/details/com.orama_interactive.Pixelorama).

**Comment by O-XiXu** [comment]
why user should move it around?! if he does, so he can run --register-app option again, idk
I'm not familiar with flatpak, but with such a option godot can be more standalone 
when i downloading godot from official site, i expect there be a setup option and remove setup option
it can help anyone that want the latest version from site

**Comment by Calinou** [comment]
> why user should move it around?! if he does, so he can run --register-app option again, idk

You'd be surprised sometimes :slightly_smiling_face: 
I'm fairly sure we'll get issue reports about the desktop shortcut not moving anymore because the Godot binary was moved after registering the shortcut.

**Comment by O-XiXu** [comment]
Yes, you're right, maybe I'm too lazy :)
but something, what if after registering, godot shows a warning about this?
or no,
godot zip file, in web site, can have a setup file (a bash)
with a icon and .desktop file, user can run the shell program and
godot will setup, it doesn't need to modify the godot command line options
forget it, i will try what you said and Thanks

**Comment by Calinou** [comment]
@bruvzg has figured out [a way to create application launchers directly from Godot](https://github.com/godotengine/godot/pull/58455). However, I'm still not sure about allowing the Godot editor to create a shortcut to itself, since it'll break when the binary is moved.

**Comment by akien-mga** [comment]
Yeah I don't think that's a good idea either. Aside from breaking when the user moves the binary, it will also need to be unregistered with the old version, and re-registered with the new version, whenever you update Godot (and Godot gets new versions quite frequently).

It likely wouldn't have much discoverability either, most users won't think that it could be provided as a command line option, and looking it up in the docs would take more time than manually creating a desktop menu entry using the DE provided tools.

Godot provides a desktop file for Linux distro packagers, which you can use for your own needs if you want: https://github.com/godotengine/godot/tree/master/misc/dist/linux

**Comment by Talkys** [comment]
I have another sugestion.
Godot could export a Flatpak bundle for Linux builds just like GNOME builder does.
This way is a lot easier to distribute the games and don't rely on any store.
Unifying the package format is a side benefit too.
